### PR TITLE
Module Loading

### PR DESCRIPTION
**Pull request by sp1tz**
- Remove branch as whitelist is no longer optional.
- Use `for..in` since prototype members are of no concern here.
- Expect a function and not an object. For this initial change we'll have to sync this w/ Ocelot settings, thereafter we can make changes here/there w/o having to sync.

### CODE REVIEW

**Comment by sp1tz** [comment]
https://github.com/umass-compsci220/ocelot-settings/pull/1 👀

File: ts/index.ts [hunk 1]
```diff
@@ -35,15 +35,17 @@ class ElementaryRunner implements CompileOK {
...
-    if (whitelistCode) {
-      Object.keys(whitelistCode).forEach((moduleName) => {
-        this.codeMap[moduleName] = eval(`(${whitelistCode[moduleName]})`);
-      });
+    const config: string = `{
+      getRunner: runtime.getRunner,
+      stopifyArray: runtime.stopifyArray,
+      stopifyObjectArrayRecur: runtime.stopifyObjectArrayRecur
+    }`;
+    for (const moduleName in opts.whitelistCode) {
```
Comment (arjunguha) [inline ts/index.ts]: There is general consensus that `for .. in` is a bad idea. The Mozilla article describes it mildly:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

This code assumes that no non-enumerable properties exist on `Object.prototype`, which happens to be true, but is not a very safe assumption.

Besides, in this code base (and in our research group), loops are considered bad design. Use a better-abstraction, such as higher-order functions.

File: ts/index.ts [hunk 2]
```diff
@@ -76,7 +78,7 @@ class ElementaryRunner implements CompileOK {
       }),
       require: (lib: string): any => {
         if (this.codeMap[lib]) {
-          return this.codeMap[lib];
+          return Object.freeze(this.codeMap[lib]);
```
Comment (arjunguha) [inline ts/index.ts]: I am in favor using Object.freeze. But, would you add a comment explaining what the threat is? i.e., what could a student do that warrants the `Object.freeze`?

File: ts/index.ts (hunk 1 above)
Comment (sp1tz) [reply ts/index.ts]: I'm aware of the concerns, as I made a comment in the PR notes.
I will change to use HOF.

File: ts/index.ts (hunk 2 above)
Comment (sp1tz) [reply ts/index.ts]: Do you mean a comment in the code? This isn't the first place `Object.freeze` is used.

**Comment by arjunguha** [comment]
Actually, this commit is not yet ready. It will be ready when lib220.ts is deleted from this branch (and tests that use lib220 work).

**Comment by arjunguha** [comment]
I assume lib220 tests are moved into ocelot-settings.

Anyway, this is great! Merge away! (Both this and Ocelot)

**Comment by arjunguha** [comment]
And for consistency, do redeploy this. (Though it should not have any effect.)

**Pull request merged**